tsconfig.json

```diff
@@ -9,7 +9,7 @@
     "allowUnreachableCode": false,
     "allowUnusedLabels": false,
     "esModuleInterop": true,
-    "importsNotUsedAsValues": "error",
+    "verbatimModuleSyntax": true,
```

Comment:
You will not find this update on FCM app since this change is done at the root directory and adding same to both PRs could possibly cause conflicts.

Reply:
I think the CI is failing because of this change? 

Are you planning on fixing the CI status checks before merging?

Reply:
Ah! I didn't know that updating this key could break CI. I will check it before merging.

Reply:
Looking good! This is the last unresolved conversation and it appears the CI status checks are still failing. 

I approve of all the changes done thus far. Assuming that more code will need to be changed and then reviewed to get the CI fixed, I will wait to approve.

Reply:
@levibostian By any chance you know what version of typescript is CI using ? `verbatimModuleSyntax` is introduced in Typescript 5 and we seem not be pushing `package-lock.json` file for the sample apps.

Reply:
[The CI is failing when it runs yarn on the root directory of the RN SDK](https://github.com/customerio/customerio-reactnative/actions/runs/8708529880/job/23886112598?pr=257#step:4:1), not running yarn in the sample app subdirectories. 

I do see that you updated the `package.json` file in this PR but the `yarn.lock` file didn't get updated. That might be the issue. Updating the yarn.lock file and pushing that up might fix this issue.